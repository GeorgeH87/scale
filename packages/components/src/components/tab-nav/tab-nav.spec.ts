/**
 * @license
 * Scale https://github.com/telekom/scale
 *
 * Copyright (c) 2021 Egor Kirpichev and contributors, Deutsche Telekom AG
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */

import { newSpecPage } from '@stencil/core/testing';
import { TabNav } from './tab-nav';

describe('TabNav', () => {
  let page;
  beforeEach(async () => {
    page = await newSpecPage({
      components: [TabNav],
      html: `<scale-tab-nav></scale-tab-nav>`,
    });
  });

  it('should match snapshot', async () => {
    expect(page.root).toMatchSnapshot();
  });

  // This class seems silly but it's needed in order to make sure we query the element
  // since we can never be certain the name of the element will be <scale-tab-nav>.
  it('should have default scale-tab-nav class', async () => {
    expect(page.root.classList.contains('scale-tab-nav')).toBe(true);
  });
});
