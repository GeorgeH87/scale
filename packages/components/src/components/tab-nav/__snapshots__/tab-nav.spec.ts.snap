// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabNav should match snapshot 1`] = `
<scale-tab-nav class="scale-tab-nav" role="tablist">
  <mock:shadow-root>
    <div class="tab-nav tab-nav--" part="tab-nav">
      <slot name="tab"></slot>
      <slot name="panel"></slot>
    </div>
  </mock:shadow-root>
</scale-tab-nav>
`;
